src/selectors/linked-nodes.test.js

```diff
@@ -5,20 +5,20 @@ import reducer from '../reducers';
 
 describe('getLinkedNodes function', () => {
   it('should search through edges for ancestor and descendant nodes', () => {
-    const { nodes } = mockState.animals.graph;
-    const nodeID = nodes.find(d => d.id.includes('salmon')).id;
-    const newMockState = reducer(mockState.animals, toggleNodeClicked(nodeID));
+    const { nodes } = mockState.testData.graph;
+    const nodeID = nodes.find(d => d.id.includes('4ffcf321')).id;
+    const newMockState = reducer(mockState.testData, toggleNodeClicked(nodeID));
     const linkedNodes = getLinkedNodes(newMockState);
     expect(linkedNodes).toEqual(expect.any(Object));
-    expect(linkedNodes['task/salmon']).toBe(true);
-    expect(linkedNodes['data/pig']).toBe(true);
-    expect(linkedNodes['task/trout']).toBe(true);
-    expect(linkedNodes['data/whale']).toBe(true);
-    expect(linkedNodes['data/horse']).toBe(true);
-    expect(linkedNodes['data/sheep']).toBe(true);
-    expect(linkedNodes['data/cat']).toBe(true);
-    expect(linkedNodes['data/dog']).toBe(true);
-    expect(linkedNodes['data/parameters']).toBe(true);
-    expect(linkedNodes['data/parameters_rabbit']).toBe(true);
+    expect(linkedNodes['4ffcf321']).toBe(true);
+    expect(linkedNodes['2cd4ba93']).toBe(true);
+    expect(linkedNodes['7196f150']).toBe(true);
+    expect(linkedNodes['1769e230']).toBe(true);
+    expect(linkedNodes['091b5035']).toBe(true);
+    expect(linkedNodes['6525f2e6']).toBe(true);
+    expect(linkedNodes['9d989e8d']).toBe(true);
+    expect(linkedNodes['e4951252']).toBe(true);
+    expect(linkedNodes['f1f1425b']).toBe(true);
+    expect(linkedNodes['2a31900d']).toBe(true);
```

Comment:
This is tricky, because these are no longer very identifiable. I would be tempted to put them into  a  lookup table of names and IDs so you can easily see from the test which is which, e.g.
```es6
describe('getLinkedNodes function', () => {
  const { nodes } = mockState.animals.graph;
  const nodeID = nodes.find(d => d.id.includes('salmon')).id;
  const newMockState = reducer(mockState.animals, toggleNodeClicked(nodeID));
  const linkedNodes = getLinkedNodes(newMockState);

  it('should return an object', () => {
    expect(linkedNodes).toEqual(expect.any(Object));
  });

  describe('should return true for ancestor/descendant nodes', () => {
    test.each([
      ['salmon', '4ffcf321'],
      ['pig', '2cd4ba93'],
      ['trout', '7196f150'],
      ['whale', '1769e230'],
      ['horse', '091b5035'],
      ['sheep', '6525f2e6'],
      ['cat', '9d989e8d'],
      ['dog', 'e4951252'],
      ['parameters', 'f1f1425b'],
      ['parameters_rabbit', '2a31900d']
    ])('node %s should be true', (name, id) => {
      expect(linkedNodes[id]).toBe(true);
    });
  });
});
```

It might be worth adding a new separate test for the 'false' case too. I'm sorry, it's not a very good test and that's my fault. I'm happy to work with you on this!

Reply:
great call, I have the same feeling when I was modifying the test as well, currently it's extremely unreadable and those without context of this test will have no clue on what's going on - I will modify it to set it into a table :)